include/vcpkg/base/strings.h

```diff
@@ -137,24 +124,32 @@ namespace vcpkg::Strings
     template<class Container, class Transformer>
     std::string join(StringLiteral delimiter, const Container& v, Transformer transformer)
     {
-        const auto begin = std::begin(v);
-        const auto end = std::end(v);
-
-        return join(delimiter, begin, end, transformer);
+        return join(delimiter, std::begin(v), std::end(v), transformer);
     }
 
     template<class InputIterator>
-    std::string join(StringLiteral delimiter, InputIterator begin, InputIterator end)
+    std::string join(StringLiteral delimiter, InputIterator first, InputIterator last)
```

Comment:
Ditto NRVO. Did this make a perf difference? IIRC I used a lambda when I did this because I didn't measure any difference....